python/ray/_private/monitor.py

```diff
@@ -116,6 +121,9 @@ def __init__(self,
 
         logger.info("Monitor: Started")
 
+    def __del__(self):
+        disconnect_to_gcs(self.gcs_client)
```

Comment:
What happens if the GCS client is garbage collected without disconnecting? Couldn't that disconnecting logic go into the `__dealloc__` method in the Cython `GcsClient` class or something like that?

Reply:
That's OK, the desctructor will be called directly. I believe I already have destructor there.

This del is actually a little bit tricky. Here, the owner of the client is monitor, but it's set in another place (core_worker). When we stop the system, we don't want another thread is referencing gcs which will give seg fault. 

This is for a safe stop.